src/client.rs

```diff
@@ -706,30 +706,42 @@ impl<T: ClientTransport + 'static> CoAPClient<T> {
         mut handler: H,
     ) -> IoResult<oneshot::Sender<ObserveMessage>> {
         let this = self.clone();
-        let mut register_packet = request;
+        let mut register_packet = request.clone();
         if 0 == register_packet.message.header.message_id {
             register_packet.message.header.message_id = self.gen_message_id();
         }
+        if register_packet.message.get_token().is_empty() {
+            register_packet
+                .message
+                .set_token(self.gen_message_id().to_be_bytes().to_vec());
```

Comment:
What is the purpose of setting default token by message id

Reply:
Right. I used gen_message_id to generate a random value for the token. I should have probably made a separate function for that.

Reply:
Great. I think a separate function is better. The message id is not a random value.